OpenRA.Game/Graphics/SpriteRenderable.cs

```diff
@@ -94,21 +116,42 @@ public void Render(WorldRenderer wr)
 			if ((tintModifiers & TintModifiers.ReplaceColor) != 0)
 				a *= -1;
 
-			wsr.DrawSprite(sprite, palette, ScreenPosition(wr), scale, t, a);
+			wsr.DrawSprite(sprite, palette, ScreenPosition(wr), scale, t, a, rotation.RendererRadians());
 		}
 
 		public void RenderDebugGeometry(WorldRenderer wr)
 		{
 			var pos = ScreenPosition(wr) + sprite.Offset;
 			var tl = wr.Viewport.WorldToViewPx(pos);
 			var br = wr.Viewport.WorldToViewPx(pos + sprite.Size);
-			Game.Renderer.RgbaColorRenderer.DrawRect(tl, br, 1, Color.Red);
+			if (rotation == WAngle.Zero)
+				Game.Renderer.RgbaColorRenderer.DrawRect(tl, br, 1, Color.Red);
+			else
+				Game.Renderer.RgbaColorRenderer.DrawPolygon(Util.RotateQuad(tl, br - tl, rotation.RendererRadians()), 1, Color.Red);
 		}
 
 		public Rectangle ScreenBounds(WorldRenderer wr)
 		{
 			var screenOffset = ScreenPosition(wr) + sprite.Offset;
-			return new Rectangle((int)screenOffset.X, (int)screenOffset.Y, (int)sprite.Size.X, (int)sprite.Size.Y);
+			if (rotation == WAngle.Zero)
+				return new Rectangle((int)screenOffset.X, (int)screenOffset.Y, (int)sprite.Size.X, (int)sprite.Size.Y);
+			else
+			{
+				var rotatedQuad = Util.RotateQuad(screenOffset, sprite.Size, rotation.RendererRadians());
+				var minX = rotatedQuad[0].X;
+				var maxX = rotatedQuad[0].X;
+				var minY = rotatedQuad[0].Y;
+				var maxY = rotatedQuad[0].Y;
+				for (var i = 0; i < rotatedQuad.Length; i++)
+				{
+					minX = rotatedQuad[i].X < minX ? rotatedQuad[i].X : minX;
+					maxX = rotatedQuad[i].X > maxX ? rotatedQuad[i].X : maxX;
+					minY = rotatedQuad[i].Y < minY ? rotatedQuad[i].Y : minY;
+					maxY = rotatedQuad[i].Y > maxY ? rotatedQuad[i].Y : maxY;
+				}
+
+				return new Rectangle((int)minX, (int)minY, (int)(maxX - minX), (int)(maxY - minY));
```

Comment:
Good point about needing Math.Ceiling, I have since made the change. Re: the rectangle having an exclusive upper bound however, I had a look and it doesn't seem to be the case, e.g. `Rectangle.Right` is `X + Width`, `Rectangle.Bottom` is `Y + Height`, which would produce the correct result even with a Width of 1 or Height of 1. Unless other parts of the rendering code are doing `Rectangle.Right - 1`, or `Rectangle.Bottom - 1` (admittedly I have not looked as I am not too familiar with how ScreenBounds() is used. Also I omitted the `(int)` on `minX` and `minY` since this would lead to a further +1 added to the width and height, e.g. `Math.Ceiling(1.1 - 0.9)` would become `Math.Ceiling(1.1 - 0)`, making an object with a width of 1 return 2 for example. If we need to pad these values I think it's better to simply add +1, +2, or whatever the padding is required, but based on how this was coded previously by passing `size` with no padding, I don't think it is necessary.

Reply:
I would need to verify the Rectangle bounds, so let's ignore the +1 faff on that for now. The `(int)` cast is needed though. The 0.9-1.1 example has a width of only 0.2, but the bounding area is still 0 to 2. A result of 0 to 1 means we're slicing off some of the area!

Reply:
Ah good point, I see the issue now. Even though a Width of `1` should normally cover a range between 0.9 to 1.1, the `minX` is actually 0 because `minX` is being cast from 0.9 to 0 via `(int)minX`. Therefore from a starting point of 0 a Width of `2` is needed to cover the point 1.1. I have refactored the above slightly to make it easier to understand, by keeping the `(int)minX` outside of the `Ceiling()` function, to make it easier to draw the parallel that I initially missed between this `(int)minX` and the first parameter `(int)minX` passed to this Rectangle function.